src/Explorer/Controls/Settings/SettingsSubComponents/MongoIndexingPolicy/MongoIndexingPolicyComponent.tsx

```diff
@@ -0,0 +1,385 @@
+    mongoIndex: MongoIndex,
+    arrayPosition: number,
+    isCurrentIndex: boolean
+  ): MongoIndexDisplayProps => {
+    const keys = mongoIndex.key.keys;
+    const type = getMongoIndexType(keys);
+    const definition = keys.join();
+    let mongoIndexDisplayProps: MongoIndexDisplayProps;
+    if (type) {
+      mongoIndexDisplayProps = {
+        definition: <Text>{definition}</Text>,
+        type: <Text>{getMongoIndexTypeText(type)}</Text>,
+        actionButton: definition === MongoIndexIdField ? <></> : this.getActionButton(arrayPosition, isCurrentIndex)
+      };
+    }
+    return mongoIndexDisplayProps;
+  };
+
+  private renderIndexesToBeAdded = (): JSX.Element => {
+    const indexesToAddLength = this.props.indexesToAdd.length;
+    for (let i = 0; i < indexesToAddLength; i++) {
+      const existingIndexToAddRef = React.createRef<AddMongoIndexComponent>();
+      this.addMongoIndexComponentRefs[i] = existingIndexToAddRef;
+    }
+    const newIndexToAddRef = React.createRef<AddMongoIndexComponent>();
+    this.addMongoIndexComponentRefs[indexesToAddLength] = newIndexToAddRef;
+
+    return (
+      <Stack {...addMongoIndexStackProps} styles={mediumWidthStackStyles}>
+        {this.props.indexesToAdd.map((mongoIndexWithType: AddMongoIndexProps, arrayPosition: number) => {
+          const keys = mongoIndexWithType.mongoIndex.key.keys;
+          const type = mongoIndexWithType.type;
+          const notification = mongoIndexWithType.notification;
+          return (
+            <AddMongoIndexComponent
+              ref={this.addMongoIndexComponentRefs[arrayPosition]}
+              position={arrayPosition}
+              key={arrayPosition}
+              description={keys.join()}
+              type={type}
+              notification={notification}
+              onIndexAddOrChange={(description: string, type: MongoIndexTypes) =>
+                this.props.onIndexAddOrChange(arrayPosition, description, type)
+              }
+              onDiscard={() => {
+                this.addMongoIndexComponentRefs.splice(arrayPosition, 1);
+                this.props.onRevertIndexAdd(arrayPosition);
+              }}
+            />
+          );
+        })}
+
+        <AddMongoIndexComponent
+          ref={this.addMongoIndexComponentRefs[indexesToAddLength]}
+          disabled={this.isIndexingTransforming()}
+          position={indexesToAddLength}
+          key={indexesToAddLength}
+          description={undefined}
+          type={undefined}
+          notification={undefined}
+          onIndexAddOrChange={(description: string, type: MongoIndexTypes) =>
+            this.props.onIndexAddOrChange(indexesToAddLength, description, type)
+          }
+          onDiscard={() => {
+            this.props.onRevertIndexAdd(indexesToAddLength);
+          }}
+        />
+      </Stack>
+    );
+  };
+
+  private renderInitialIndexes = (): JSX.Element => {
+    const initialIndexes: MongoIndexDisplayProps[] = this.props.mongoIndexes
```

Comment:
There are a lot of places you are specifying types where you don't need to. The TS compiler can infer from usage. Example here:

```  ts
const initialIndexes: MongoIndexDisplayProps[] = this.props.mongoIndexes
      .map((mongoIndex: MongoIndex, arrayPosition: number) =>
        this.getMongoIndexDisplayProps(mongoIndex, arrayPosition, true)
      )
      .filter(
        (value: MongoIndexDisplayProps, arrayPosition: number) =>
          !!value && !this.props.indexesToDrop.includes(arrayPosition)
      );
```

Without types

```  ts
const initialIndexes = this.props.mongoIndexes
      .map((mongoIndex, arrayPosition) =>
        this.getMongoIndexDisplayProps(mongoIndex, arrayPosition, true)
      )
      .filter(
        (value, arrayPosition) =>
          !!value && !this.props.indexesToDrop.includes(arrayPosition)
      );
```

TS will tell you if it is trying to infer `any` and fail to compile

Reply:
will remove